### PR TITLE
[mod_login] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8')

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #10399 .
#### Summary of Changes

Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8')
#### Testing Instructions
- Enable the mod_login module to the frontend
- see that it works (login)
- see that it works (logout)
- apply this patch
- see that it works (login)
- see that it works (logout)

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 8a3e6636893d1d73bb1c20dd3e1aa3d0a3e4de4f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10406">issues.joomla.org/joomla-cms/10406</a>.</sub>

**Comment by BurtNL** [comment]
I have tested this item :white_check_mark: successfully on 8a3e6636893d1d73bb1c20dd3e1aa3d0a3e4de4f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10406">issues.joomla.org/joomla-cms/10406</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10406">issues.joomla.org/joomla-cms/10406</a>.</sub>

File: modules/mod_login/tmpl/default_logout.php [hunk 1]
```diff
@@ -11,14 +11,14 @@
...
 <?php if ($params->get('greeting')) : ?>
 	<div class="login-greeting">
-	<?php if ($params->get('name') == 0) : {
-		echo JText::sprintf('MOD_LOGIN_HINAME', htmlspecialchars($user->get('name')));
-	} else : {
-		echo JText::sprintf('MOD_LOGIN_HINAME', htmlspecialchars($user->get('username')));
-	} endif; ?>
+	<?php if ($params->get('name') == 0) : ?>
+		<?php echo JText::sprintf('MOD_LOGIN_HINAME', htmlspecialchars($user->get('name')), ENT_COMPAT, 'UTF-8'); ?>
+	<?php else : ?>
```
Comment (andrepereiradasilva) [inline modules/mod_login/tmpl/default_logout.php]: Parentheses is in the wrong place.

File: modules/mod_login/tmpl/default_logout.php [hunk 2]
```diff
@@ -11,14 +11,14 @@
...
+		<?php echo JText::sprintf('MOD_LOGIN_HINAME', htmlspecialchars($user->get('username')), ENT_COMPAT, 'UTF-8'); ?>
```
Comment (andrepereiradasilva) [inline modules/mod_login/tmpl/default_logout.php]: Parentheses

**Comment by andrepereiradasilva** [comment]
@brianteeman same thing. Please remove the RTC.

**Comment by wojsmol** [comment]
@zero-24 @andrepereiradasilva Please see https://github.com/zero-24/joomla-cms/issues/13

**Comment by andrepereiradasilva** [comment]
seems fine to me now.

**Comment by zero-24** [comment]
Back to Pending. Thanks @wojsmol and @andrepereiradasilva <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10406">issues.joomla.org/joomla-cms/10406</a>.</sub>

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on fd6e8774c2ccee1bf2fa87164721f7a0c0ec19af<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10406">issues.joomla.org/joomla-cms/10406</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on fd6e8774c2ccee1bf2fa87164721f7a0c0ec19af<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10406">issues.joomla.org/joomla-cms/10406</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10406">issues.joomla.org/joomla-cms/10406</a>.</sub>

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**